spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/feature/filters/TargetingFilter.java

```diff
@@ -0,0 +1,191 @@
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager.feature.filters;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.spring.cloud.feature.manager.FeatureFilter;
+import com.microsoft.azure.spring.cloud.feature.manager.TargetingException;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureFilterEvaluationContext;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.Audience;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.GroupRollout;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.ITargetingContextAccessor;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.TargetingContext;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.TargetingEvaluationOptions;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.TargetingFilterSettings;
+
+public class TargetingFilter implements FeatureFilter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TargetingFilter.class);
+    
+    private static final String USERS = "users";
+    private static final String GROUPS = "groups";
+    private static final String AUDIENCE = "Audience";
+    
+    private static final String OUT_OF_RANGE = "The value is out of the accepted range.";
+    private static final String REQUIRED_PARAMETER = "Value cannot be null.";
+
+    private ITargetingContextAccessor contextAccessor;
+
+    private TargetingEvaluationOptions options;
+    
+    public TargetingFilter(ITargetingContextAccessor contextAccessor) {
+        this.contextAccessor = contextAccessor;
+        this.options = new TargetingEvaluationOptions();
+    }
+
+    public TargetingFilter(ITargetingContextAccessor contextAccessor, TargetingEvaluationOptions options) {
+        this.contextAccessor = contextAccessor;
+        this.options = options;
+    }
+
+    @Override
+    public boolean evaluate(FeatureFilterEvaluationContext context) {
+        if (context == null) {
+            throw new IllegalArgumentException("Targeting Context not configured.");
+        }
+
+        TargetingContext targetingContext = contextAccessor.getContextAsync().block();
+
+        if (targetingContext == null) {
+            LOGGER.warn("No targeting context available for targeting evaluation.");
+            return false;
+        }
+
+        TargetingFilterSettings settings = new TargetingFilterSettings();
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
```

Comment:
It would be a good idea to add this objectMapper as static final once this one is thread-safe since the configuration doesn't be changed.
```java
       private static final ObjectMapper mapper = new ObjectMapper()
            .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
```